ts/features/pn/loginEngagement/hooks/useSendActivationFlow.tsx

```diff
@@ -7,16 +7,27 @@ import { useIONavigation } from "../../../../navigation/params/AppParamsList";
 import { setSendEngagementScreenHasBeenDismissed } from "../store/actions";
 import PN_ROUTES from "../../navigation/routes";
 import { MESSAGES_ROUTES } from "../../../messages/navigation/routes";
+import { areNotificationPermissionsEnabledSelector } from "../../../pushNotifications/store/reducers/environment";
+import { NOTIFICATIONS_ROUTES } from "../../../pushNotifications/navigation/routes";
 import { setSecurityAdviceReadyToShow } from "../../../authentication/fastLogin/store/actions/securityAdviceActions";
 
 export const useSendActivationFlow = () => {
-  const { pop, replace } = useIONavigation();
+  const { popToTop, replace } = useIONavigation();
   const dispatch = useIODispatch();
   const toast = useIOToast();
   const isActivating = useIOSelector(isLoadingPnActivationSelector);
+  const notificationPermissionsEnabled = useIOSelector(
+    areNotificationPermissionsEnabledSelector
+  );
 
   const onSENDActivationSucceeded = () => {
-    pop();
+    if (notificationPermissionsEnabled) {
```

Comment:
It's not related to this PR but, being the dispatch of actions a sequential and synchronous operation, I think that it is better to first do
```
dispatch(setSendEngagementScreenHasBeenDismissed());
dispatch(setSecurityAdviceReadyToShow(true));
``` 
and then the `if { ... } else { ... }` for the navigation